### PR TITLE
rename apiserver_storage_object_counts to apiserver_storage_objects

### PR DESCRIPTION
**Pull request by logicalhan**
To conform to instrumentation best practices. 

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Pluralized `_counts` is actually an anti-pattern for metrics, instead we pluralize the name of the thing being counted for gauges. 

#### Does this PR introduce a user-facing change?

```release-note
`apiserver_storage_objects` (a newer version of `etcd_object_counts) is promoted and marked as stable. 
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1209-metrics-stability

```

### CODE REVIEW

**Review by erain — APPROVED** [review]
LGTM, and thanks!!

**Comment by ehashman** [comment]
Are we just waiting on the exception request approval to remove the hold?

**Comment by logicalhan** [comment]
> Are we just waiting on the exception request approval to remove the hold?

yes.

**Comment by palnabarun** [comment]
/milestone v1.21

Putting back the milestone since the exception request was APPROVED. ref: https://groups.google.com/g/kubernetes-sig-release/c/p9E6s14269M/m/f0L7VhbeAgAJ

**Comment by ehashman** [comment]
/hold cancel

https://groups.google.com/g/kubernetes-sig-release/c/p9E6s14269M/m/2V8WoebdAgAJ

**Pull request merged**
